lib/db_utils.go

```diff
@@ -1643,6 +1731,546 @@ func DBGetAllUserGroupEntries(handle *badger.DB, ownerPublicKey []byte) ([]*Mess
+		return false, nil
+	} else if err != nil {
+		return false, errors.Wrapf(err, "DBGetGroupMemberForAccessGroupWithTxn: Problem getting group member for access group"+
+			" groupOwnerPublicKey: %v, groupKeyName: %v, groupMemberPublicKey: %v", groupOwnerPublicKey, groupKeyName, groupMemberPublicKey)
+	}
+
+	return true, nil
+}
+
+func DBGetPaginatedAccessGroupMembersFromEnumerationIndex(handle *badger.DB, snap *Snapshot,
+	groupOwnerPublicKey PublicKey, groupKeyName GroupKeyName, startingAccessGroupMemberPublicKeyBytes []byte, limit uint32) (
+	_accessGroupMemberPublicKeys []*PublicKey, _err error) {
+
+	var err error
+	var accessGroupMemberPublicKeys []*PublicKey
+	err = handle.View(func(txn *badger.Txn) error {
+		accessGroupMemberPublicKeys, err = DBGetPaginatedAccessGroupMembersFromEnumerationIndexWithTxn(txn, snap,
+			groupOwnerPublicKey, groupKeyName, startingAccessGroupMemberPublicKeyBytes, limit)
+		return err
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "DBGetPaginatedAccessGroupMembersFromEnumerationIndex: "+
+			"Problem getting paginated access group members from enumeration index with "+
+			"groupOwnerPublicKey (%v) groupKeyName (%v) startGroupMemberPublicKey (%v) limit (%v)",
+			groupOwnerPublicKey, groupKeyName, startingAccessGroupMemberPublicKeyBytes, limit)
+	}
+	return accessGroupMemberPublicKeys, nil
+}
+
+// DBGetPaginatedAccessGroupMembersFromEnumerationIndexWithTxn for a given group will return a list of group members,
+// with public keys lexicographically GREATER than startingAccessGroupMemberPublicKeyBytes. The number of returned entries
+// will be at most maxMembersToFetch.
+func DBGetPaginatedAccessGroupMembersFromEnumerationIndexWithTxn(txn *badger.Txn, snap *Snapshot,
+	groupOwnerPublicKey PublicKey, groupKeyName GroupKeyName, startingAccessGroupMemberPublicKeyBytes []byte,
+	maxMembersToFetch uint32) (_accessGroupMemberPublicKeys []*PublicKey, _err error) {
+
+	var accessGroupMembers []*PublicKey
+	prefix := _dbSeekPrefixForAccessGroupMemberEnumerationIndex(groupOwnerPublicKey, groupKeyName)
+	startKey := append(prefix, startingAccessGroupMemberPublicKeyBytes...)
+
+	keysFound := _enumeratePaginatedLimitedKeysForPrefixWithTxn(txn, prefix, startKey, maxMembersToFetch)
+	// Sanity-check that we don't return the starting key.
+	if len(keysFound) > 0 && bytes.Compare(startKey, keysFound[0]) == 0 {
+		// We will fetch one more key after the last key found and append it to the list of keys found, with the first element removed.
+		additionalKey := _enumeratePaginatedLimitedKeysForPrefixWithTxn(txn, prefix, keysFound[len(keysFound)-1], 2)
+		keysFound = append(keysFound[1:], additionalKey[1:]...)
+	}
+
+	for _, key := range keysFound {
+		memberPublicKeyBytes := key[len(prefix):]
+		accessGroupMembers = append(accessGroupMembers, NewPublicKey(memberPublicKeyBytes))
+	}
+
+	// Sanity-check that the public keys we found are actually lexicographically sorted.
+	for ii := 1; ii < len(accessGroupMembers); ii++ {
```

Comment:
this seems unnecessary since badger is sorted and these compares could be costly.